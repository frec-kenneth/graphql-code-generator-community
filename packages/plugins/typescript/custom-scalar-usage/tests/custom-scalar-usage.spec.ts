import { buildSchema } from 'graphql';
import { mergeOutputs } from '@graphql-codegen/plugin-helpers';
import { plugin } from '../src/index.js';

describe('custom-scalar-usage', () => {
  it('Should output usage object correctly', async () => {
    const schema = buildSchema(/* GraphQL */ `
      scalar DateTime
      scalar LocalDate
      type Query {
        user: User!
      }
      type User {
        id: ID!
        name: String!
        signupDate: LocalDate
        creationDate: DateTime!
        logins: [DateTime]
        referrals: [DateTime!]!
      }
    `);
    const result = mergeOutputs([await plugin(schema, [], { scalars: ['DateTime', 'LocalDate'] })]);
    expect(result).toMatchSnapshot();
  });

  it('Should only output specified types', async () => {
    const schema = buildSchema(/* GraphQL */ `
      scalar DateTime
      type Query {
        user: User!
      }
      type User {
        id: ID!
        name: String!
        signupDate: DateTime
        creationDate: DateTime!
        logins: [DateTime]
        referrals: [DateTime!]!
      }
    `);
    const result = mergeOutputs([await plugin(schema, [], { scalars: ['LocalDate'] })]);
    expect(result).toMatchSnapshot();
  });
});
