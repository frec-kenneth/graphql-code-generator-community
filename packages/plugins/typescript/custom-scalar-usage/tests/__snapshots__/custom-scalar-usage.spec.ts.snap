// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-scalar-usage Should only output specified types 1`] = `
"export const scalarUsage = {
  "LocalDate": {}
}"
`;

exports[`custom-scalar-usage Should output usage object correctly 1`] = `
"export const scalarUsage = {
  "DateTime": {
    "User": [
      "creationDate",
      "logins",
      "referrals"
    ]
  },
  "LocalDate": {
    "User": [
      "signupDate"
    ]
  }
}"
`;
